src/App.js

```diff
@@ -1,85 +1,107 @@
-            <input type="text" className="edit" />
-          </li>
+  handlFilter = (string) => {
+    const str = string;
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    this.setState(() => ({
+      todoFilter: str,
+    }));
+  }
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-4" />
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  deleteTodo = (id) => {
+    this.setState(state => ({
+      todos: state.todos.filter(item => item.id !== id),
+    }));
+  }
+
+  deleteCompleted = (id) => {
+    this.setState(state => ({
+      todos: state.todos.filter(item => !item.completed),
+    }));
+  }
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
+  markAll = () => {
+    this.setState(state => ({
+      todos: state.todos.map(item => ({
+        ...item,
+        completed: state.toggleAll,
+      })),
+      toggleAll: !state.toggleAll,
+    }));
+  }
 
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
+  render() {
+    let todos = [];
 
-          <li>
-            <a href="#/active">Active</a>
-          </li>
+    if (this.state.todoFilter === 'All') {
+      todos = [...this.state.todos];
+    } else if (this.state.todoFilter === 'Active') {
+      todos = this.state.todos.filter(item => !item.completed);
+    } else if (this.state.todoFilter === 'Completed') {
+      todos = this.state.todos.filter(item => item.completed);
+    }
 
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
+    return (
+      <section className="todoapp">
+        <TodoApp
+          addTodo={this.addTodo}
+        />
+
+        <section className="main">
+          {todos
+            && (
+              <TodoList
+                items={todos}
+                doneTask={this.doneTask}
+                deleteTodo={this.deleteTodo}
+                markAll={this.markAll}
+              />
+            )}
+        </section>
+        {this.state.todos.filter(item => item).length >= 1 ? (
+          <TodosFilter
+            item={todos.filter(item => !item.completed).length}
+            completed={todos.filter(item => item.completed).length}
+            handlFilter={this.handlFilter}
+            filter={this.state.todoFilter}
+            deleteCompleted={this.deleteCompleted}
+          />
+        ) : ''}
```

Comment:
```suggestion
        )}
```
Fix it in all places